File: springboot/security/basic-board_token/src/main/java/com/example/basicboard_token/domain/entity/Board.java
package com.example.basicboard_token.domain.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "board")
@Getter
@AllArgsConstructor(access = AccessLevel.PROTECTED)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Board {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false, length = 200)
    private String title;
    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;
    @Column(nullable = false, length = 50)
    private String userId;
    @Column(length = 255)
    private String filePath;

    @Column(nullable = false, length = 200)
    private LocalDateTime createdAt;

    @OneToMany(mappedBy = "board")
    private List<Comment> comments = new ArrayList<>();

    public void update(String title, String content, String filePath) {
        this.title = title;
        this.content = content;
        this.filePath = filePath;
    }

    public static Board from(
            String title,
            String content,
            String userId,
            String filePath,
            List<Comment> comments
    ) {
        return new Board(
                null,
                title,
                content,
                userId,
                filePath,
                LocalDateTime.now(),
                comments
        );
    }
    public static Board from(
            String title,
            String content,
            String userId,
            String filePath
    ) {
        return new Board(
                null,
                title,
                content,
                userId,
                filePath,
                LocalDateTime.now(),
                new ArrayList<>()
        );
    }
}

File: kotlin/kotlin-part-4/src/main/kotlin/org/example/kotlinpart4/domain/entity/Board.kt
package org.example.kotlinpart4.domain.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.LocalDateTime

@Entity
@Table(name = "board")
class Board (

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false, length = 200)
    var title: String,

    @Column(nullable = false, columnDefinition = "TEXT")
    var content: String,

    @Column(nullable = false, length = 50)
    var userId: String,

    @Column(nullable = false)
    var created: LocalDateTime = LocalDateTime.now()
) {

    fun update(title: String, content: String) {
        this.title = title
        this.content = content
    }

}